test/src/test/java/org/corfudb/infrastructure/log/StreamLogFilesTest.java

```diff
@@ -125,4 +125,42 @@ public void testStreamLogDataCorruption() throws Exception {
         StreamLog log3 = new StreamLogFiles(logDir, false);
         assertThat(log3.read(0)).isNull();
     }
+
+    @Test
+    public void multiThreadedReadWrite() throws Exception {
+        String logDir = getDirPath();
+        StreamLog log = new StreamLogFiles(logDir, false);
+
+        ByteBuf b = ByteBufAllocator.DEFAULT.buffer();
+        byte[] streamEntry = "Payload".getBytes();
+        Serializers.CORFU.serialize(streamEntry, b);
+
+        Thread t1 = new Thread() {
```

Comment:
Can you use the scheduleConcurrently/executeScheduled framework? See SMRMapTest::concurrentAbortTest